src/Pester.Runtime.psm1

```diff
@@ -184,6 +185,7 @@ function New-Block {
     $block.Path = $Path
     $block.Tag = $Tag
     $block.ScriptBlock = $ScriptBlock
+    $block.StartLine = $StartLine
```

Comment:
Maybe the $StartLine should be nullable and fallback to $ScriptBlock.StartPosition.StartLine when the value is not provided? That way I am way less likely to run into this when I implement some other LineFilter and test it in the Runtime, but not RSpec implementation? 

Another option would be to fallback to $MyInvocation of the New-* command, but that seems like a worse thing to do.